src/App.tsx

```diff
@@ -1,24 +1,211 @@
+import { SortTodoBy } from './types/SortTodoBy';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [sortTodosBy, setSortTodosBy] = useState(SortTodoBy.Default);
+  const [loadingTodosIds, setLoadingTodosIds] = useState<number[]>([]);
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [loadingError, setLoadingError] = useState(false);
+  const [errorText, setErrorText] = useState('');
+
+  const [disableField, setDisableField] = useState(false);
+
+  const visiableTodos = filterTodos(todos, sortTodosBy);
+  const completedTodos = useCallback(() => {
+    return todos.filter(todo => todo.completed);
+  }, [todos]);
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const hasActiveTodos = activeTodos.length !== 0;
+  const hasCompletedTodos = completedTodos().length !== 0;
+
+  const handleShowError = (text: string) => {
+    setLoadingError(true);
+    setErrorText(text);
+
+    setTimeout(() => {
+      setLoadingError(false);
+    }, 3000);
+  };
+
+  const handleCloseError = () => {
+    setLoadingError(false);
+  };
+
+  const loadTodos = (userId: number) => {
+    setIsLoading(true);
+    setLoadingError(false);
+
+    getTodos(userId)
+      .then(loadedTodos => setTodos(loadedTodos))
+      .catch(() => {
+        handleShowError(TodosError.FailedLoadingGoods);
+      })
+      .finally(() => setIsLoading(false));
+  };
+
+  useEffect(() => {
+    loadTodos(USER_ID);
+  }, []);
+
+  const handleAddTodo = async (title: string) => {
+    if (!title.trim()) {
+      handleShowError(TodosError.InvalidTitle);
+    } else {
+      const newTodo = {
+        title,
+        userId: USER_ID,
+        completed: false,
+      };
+
+      try {
+        setDisableField(true);
+        setTempTodo({
+          ...newTodo,
+          id: 0,
+        });
+
+        await postTodo(newTodo)
+          .then((response) => setTodos([...todos, response]));
+      } catch {
+        handleShowError(TodosError.AddTodo);
+      } finally {
+        setDisableField(false);
+        setTempTodo(null);
+      }
+    }
+  };
+
+  const handleDeleteTodo = async (todoId: number) => {
+    try {
+      setLoadingTodosIds(prevTodos => [...prevTodos, todoId]);
+
+      await deleteTodo(todoId);
+
+      setTodos(() => todos.filter(todo => todo.id !== todoId));
+    } catch {
+      handleShowError(TodosError.DeleteTodo);
+    } finally {
+      setLoadingTodosIds([]);
+    }
+  };
+
+  const handleRemoveCompletedTodo = useCallback(() => {
+    completedTodos().forEach(({ id }) => {
+      setLoadingTodosIds((prevCompleted) => [
+        ...prevCompleted,
+        id,
+      ]);
+
+      handleDeleteTodo(id)
+        .then(() => {
+          setTodos(
+            (prevTodos) => prevTodos.filter(todo => !todo.completed),
+          );
+        })
+        .finally(() => setLoadingTodosIds([]));
```

Comment:
I guess it should be set after all todos will be removed, not after first resolver/rejected promise
or remove only id of todo that you removed here